client/controller/index.web.js

```diff
@@ -74,6 +75,16 @@ export function redirectLoggedIn( context, next ) {
 	next();
 }
 
+// Middleware for showing Calypso login form before arriving at the requested route
+export function viaLogin( context, next ) {
```

Comment:
What do you think about using a more explicit name such as `redirectWhenLoggedOut`? It would also be more consistent with the [`RedirectWhenLoggedIn` component](https://github.com/Automattic/wp-calypso/blob/5f15f3365213253724ca0d8798c6cb8160f00bc9/client/components/redirect-when-logged-in/index.jsx#L19).

Reply:
> redirectWhenLoggedOut

Yeah that sounds good 👍 . I struggled with the naming of this.